pkg/datastore/helper.go

```diff
@@ -117,6 +119,9 @@ func FromConfig(l *logrus.Entry, authorizer authorizer.Authorizer, cfg DBConfig)
 				if tx := db.gormDBMap[dbrole.MAIN].Exec(stmt); tx.Error != nil {
 					err = ErrExecutingSqlStmt.Wrap(tx.Error).WithValue(SQL_STMT, stmt)
 					db.logger.Errorln(err)
+					if strings.Contains(tx.Error.Error(), "duplicate key value") {
```

Comment:
Could you give an example of an error that would contain "duplicate key value"? Does that error happen when a Postgres user/role already exists?
Is there a way to cast the error to a relevant struct and check for one of its fields to determine if it's a "duplicate key value" error? Or, can `errors.Is()` be used to determine if the error is of the certain type?

Reply:
Look at the unit test failures, it's very rare but the initializer needs to be synchronized across multiple runs. I think it might be fine after the mutex lock